Gems/EMotionFX/Code/EMotionFX/Source/Event.h

```diff
@@ -42,7 +45,21 @@ namespace EMotionFX
         void SetEventData(size_t index, EventDataPtr&& newData);
         void InsertEventData(size_t index, EventDataPtr&& newData);
 
+        void SetEventDataChangeEvent(Event::EventDataChangeEvent::Handler& handler);
     protected:
+        struct EventContainer {
+            EventContainer() = default;
+             
+            // event container is not copyable 
+            EventContainer(const EventContainer&) {}
+            EventContainer& operator=(const EventContainer&) {
+                return *this;
+            }
```

Comment:
update: This doesn't look correct to me, these operations are not correct, should we `=delete` them if they can't be used. Currently they'll run but do the wrong thing (not actually copy anything)

Reply:
I'm cheating a bit event is not a copyable type is in stubbing the copy and assignment.

Reply:
Yeah I don't think we should do this, there definitely has to be another option

Reply:
I think in this case we should provide a proper implementation of the Event copy/move constructor and move/assignment operator to correctly clone the event and its callback, rather than this cheat. This is much like the same thing as `std::unique_ptr` which isn't copyable but you can of course do a deep copy of the underlying value if supported which I think should be possible here right? Or we need to make `Event` itself non-copyable

Reply:
I end up having the implement the copy constructor for both Event and MotionEvent I think just cheating a bit here but yes this is probably the correct thing I should of done originally.

Reply:
Yeah in cases like this the rule of 5 is the way to go (implement all special members) - you can use the `swap` idiom to implement some in terms of the others but it's up to you what you decide - this talk on the special members is really great too if you haven't seen it - https://www.youtube.com/watch?v=vLinb2fgkHk.

Reply:
yep, I ended up implementing all 5 operators.

Reply:
Okay excellent thanks @pollend, I'll take another look shortly